src/extensions/default/DebugCommands/main.js

```diff
@@ -257,6 +273,485 @@ define(function (require, exports, module) {
+        if (pref && !pref.excludeFromHints && _getObjType(pref) !== "undefined") {
+            return true;
+        }
+
+        return false;
+    }
+
+   /*
+    * This method tries to matach between initial objects
+    * and key objects and then aggregates objects from both
+    * the properties.
+    */
+    function _getObjKeys(prefObj) {
+
+        var finalObj = {},
+            property,
+            keysFound = false;
+
+        if (!prefObj) {
+            return {};
+        }
+
+        if (typeof (prefObj.initial) === "object") {
+            // iterate through the list.
+            keysFound = true;
+            for (property in prefObj.initial) {
+                if (prefObj.initial.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj.initial[property];
+                }
+            }
+        }
+
+        if (typeof (prefObj.keys) === "object") {
+            // iterate through the list.
+            var allKeys = prefObj.keys;
+            keysFound = true;
+            for (property in allKeys) {
+                if (allKeys.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj.keys[property];
+                }
+            }
+        }
+
+        // Last resort: Maybe plain objects, in which case
+        // we blindly extract all the properties.
+        if (keysFound === false) {
+            for (property in prefObj) {
+                if (prefObj.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj[property];
+                }
+            }
+        }
+
+        return finalObj;
+    }
+
+    function _formatDefault(prefObj, prefName, tabIndentStr) {
+
+        if (!prefObj || typeof (prefName) !== "string" || _getObjType(prefObj) === "object") {
+            // return empty string in case of
+            // object or pref is not defined.
+            return "";
+        }
+
+        var prefDescription = prefObj.description || "",
+            prefDefault     = prefObj.initial,
+            prefFormatText  = tabIndentStr + "\t// {0}\n" + tabIndentStr + "\t\"{1}\": {2}",
+            prefObjType     = _getObjType(prefObj);
+
+        if (prefDefault === undefined && !prefObj.description) {
+            // This could be the case when prefObj is a basic JS variable.
+            if (prefObjType === "number" || prefObjType === "boolean" || prefObjType === "string") {
+                prefDefault = prefObj;
+            }
+        }
+
+        if (prefDefault === undefined) {
+            if (prefObjType === "number") {
+                prefDefault = 0;
+            } else if (prefObjType === "boolean") {
+                // Defaulting the preference to false,
+                // in case this is missing.
+                prefDefault = false;
+            } else {
+                // for all other types
+                prefDefault = "";
+            }
+        }
+
+        if ((prefDescription === undefined || prefDescription.length === 0)) {
```

Comment:
@MarcelGerber Most of the stringent type checks that I have put in here are to make sure that we generate correct code. We definently don't want to end up with a `[object object]` strings in the default preferences file. Consider `if(!prefDescription)` is defined and is set to false. In this case, the condition evaluates to `true` while the check in the first place was to see if this was `undefined` and if so then generate a default value. So we could easily define a `description` property inside a preference and assign a `false` to it. That is good enough to break the logic. I am not saying I have handled all these type of situations, but did try best to handle whatever I can think of. :smile: